scripts/sampler/__main__.py

```diff
@@ -0,0 +1,163 @@
+    else:
+        sample_files = list(SAMPLES_DIR.glob("*.json"))
+    if not sample_files:
+        print(f"No {'matching ' if args.sample_ids else ''}sample files found.")
+        return
+
+    random.shuffle(sample_files)
+    logs = []
+    for sample_file in sample_files:
+        if args.number and len(logs) >= args.number:
+            break
+        if not sample_file.exists():
+            warn(f"Sample file {sample_file} does not exist.")
+            continue
+        sample = Sample.load(sample_file)
+        if (args.add_context or args.remove_context) and (
+            "[ADDED CONTEXT]" in sample.title or "[REMOVED CONTEXT]" in sample.title
+        ):
+            warn(f"Skipping {sample.id}: has already been modified.")
+            continue
+        if args.validate:
+            is_valid, reason = await validate_sample(sample)
+            status = (
+                "\033[92mPASSED\033[0m"
+                if is_valid
+                else f"\033[91mFAILED: {reason}\033[0m"
+            )
+            print(f"[{sample.id[:8]}] {sample.title}: {status}")
+            logs.append({"id": sample.id, "is_valid": is_valid, "reason": reason})
+        elif args.finetune:
+            print(f"Generating fine-tuning example for sample {sample.id[:8]}")
+            try:
+                example = await generate_finetune_gpt(sample)
+                example_file = FINETUNE_DIR / f"finetune_{sample.id}.json"
+                with open(example_file, "w") as f:
+                    json.dump(example, f, indent=4)
+                logs.append({"id": sample.id, "example_file": example_file})
+            except Exception as e:
+                warn(
+                    f"Error generating fine-tuning example for sample {sample.id}: {e}"
+                )
+        elif args.add_context:
+            print(f"Adding extra context to sample {sample.id[:8]}")
+            try:
+                new_sample = await add_context(sample)
+                new_sample.save(SAMPLES_DIR / f"sample_{new_sample.id}.json")
+                logs.append({"id": new_sample.id, "prototype_id": sample.id})
+            except Exception as e:
+                warn(f"Error adding extra context to sample {sample.id}: {e}")
+        elif args.remove_context:
+            if not sample.context or len(sample.context) == 1:
+                warn(f"Skipping {sample.id}: no context to remove.")
+                continue
+            try:
+                new_sample = await remove_context(sample)
+                new_sample.save(SAMPLES_DIR / f"sample_{new_sample.id}.json")
+                logs.append({"id": new_sample.id, "prototype_id": sample.id})
+            except Exception as e:
+                warn(f"Error removing context from sample {sample.id}: {e}")
+        else:
+            print(f"Evaluating sample {sample.id[:8]}")
+            print(f"  Prompt: {sample.message_prompt}")
+            diff_eval = await evaluate_sample(sample)
```

Comment:
To me `evaluate_sample` would imply running and grading the sample. Do you think `run_sample` or `execute_sample` would be better?